File: LeetCode_1513/Main.py
class Solution:
    def numSub(self, s: str) -> int:
        answer, cnt = 0, 0
        mod = 10**9 + 7
        length = len(s)
        for i in range(length):
            if s[i] == "1":
                cnt += 1
            else:
                answer += cnt * (cnt + 1) // 2
                cnt = 0

        answer += cnt * (cnt + 1) // 2
        answer %= mod
        return answer

File: LeetCode_1513/Main.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    const int mod = 1e9 + 7;
    int numSub(string s) {
        int cnt = 0;
        long long answer = 0;
        for (char ch : s) {
            if (ch == '1') {
                cnt += 1;
            } else {
                answer += 1LL * cnt * (cnt + 1) / 2;
                answer %= mod;
                cnt = 0;
            }
        }
        answer += cnt * (cnt + 1) / 2;
        answer %= mod;
        return (int)answer;
    }
};